pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -126,60 +157,18 @@ func (r *AtlasProjectReconciler) Reconcile(context context.Context, req ctrl.Req
 	return ctrl.Result{}, nil
 }
 
-func (r *AtlasProjectReconciler) Delete(e event.DeleteEvent) error {
-	project, ok := e.Object.(*mdbv1.AtlasProject)
-	if !ok {
-		r.Log.Errorf("Ignoring malformed Delete() call (expected type %T, got %T)", &mdbv1.AtlasProject{}, e.Object)
-		return nil
-	}
+func (r *AtlasProjectReconciler) deleteAtlasProject(ctx context.Context, atlasClient mongodbatlas.Client, project *mdbv1.AtlasProject) (err error) {
 	log := r.Log.With("atlasproject", kube.ObjectKeyFromObject(project))
 	log.Infow("-> Starting AtlasProject deletion", "spec", project.Spec)
 
-	if project.Status.ID == "" {
-		log.Infof("Project does not exist in Atlas, nothing to remove")
+	_, err = atlasClient.Projects.Delete(ctx, project.Status.ID)
+	var apiError *mongodbatlas.ErrorResponse
+	if errors.As(err, &apiError) && apiError.ErrorCode == atlas.NotInGroup {
+		log.Infow("Project is already deleted", "projectID", project.Status.ID)
```

Comment:
This error message is assuming something and could be misleading, please generalize and make sure entire error is logged

Reply:
Good call, changed to "Project does not exist".